doc/workshops/plugin-development.rst

```diff
@@ -2,4 +2,917 @@
+               triggered=self.stop_pomodoro_timer,
+           )
+
+And the third, add these actions in the corresponding section of the toolbar:
+
+.. code-block:: python
+
+           # Add actions to toolbar
+           self.add_item_to_toolbar(
+               start_timer_action,
+               self.pomodoro_timer_toolbar,
+               section=PomodoroToolbarSections.Controls,
+           )
+
+           self.add_item_to_toolbar(
+               pause_timer_action,
+               self.pomodoro_timer_toolbar,
+               section=PomodoroToolbarSections.Controls,
+           )
+
+           self.add_item_to_toolbar(
+               stop_timer_action,
+               self.pomodoro_timer_toolbar,
+               section=PomodoroToolbarSections.Controls,
+           )
+
+These operations, and the way they are implemented, are related to the way of programming with Qt from Python. Where we should focus our attention at this point is the proper place they take within the structure of the plugin.
+
+To review in detail the topics related to Qt we suggest the official documentation and the various tutorials that can be found on the Internet.
+
+When creating the actions, we indicate by means of the ``triggered`` parameter the methods to be executed when they are activated, i.e. when the corresponding buttons on the toolbar are clicked.
+
+We can insert these methods at the end of the ``SpyderPomodoroTimerContainer`` declaration, in the section that our cookiecutter template indicates as ``# --- Public API``.
+
+.. code-block:: python
+
+       def start_pomodoro_timer(self):
+           """Start the timer."""
+           self.pomodoro_timer_status.pause = False
+
+       def pause_pomodoro_timer(self):
+           """Pause the timer."""
+           self.pomodoro_timer_status.pause = True
+
+       def stop_pomodoro_timer(self):
+           """Stop the timer."""
+           self.pomodoro_timer_status.pause = True
+           self.pomodoro_timer_status.countdown = self.pomodoro_timer_status.pomodoro_limit
+
+These methods simply manipulate the ``pause`` field of ``pomodoro_timer_status``, and in the case of ``stop_pomodoro_timer`` the countdown is restarted.
+
+Register the toolbar
+--------------------
+
+A final mandatory step is to go to ``plugin.py`` and register this new toolbar component.
+
+To do this, add ``Plugins.Toolbar`` into our plugin requirements:
+
+.. code-block:: python
+
+       REQUIRES = [Plugins.StatusBar, Plugins.Toolbar]
+
+And connect the toolbar we have created in the container to the spyder toolbar.
+
+.. code-block:: python
+
+           toolbar = self.get_plugin(Plugins.Toolbar)
+           toolbar.add_application_toolbar(container.pomodoro_timer_toolbar)
+
+Review the changes
+------------------
+
+Something that has not been mentioned is that every time we make a change to our code we must restart Spyder so that the plugin is reloaded and we can review the changes.
```

Comment:
Probably best to mention this in the "How to test our plugin part"

Reply:
Moved to the end of "How to test our plugin part" section.